jcl/src/java.base/share/classes/java/lang/invoke/MethodHandles.java

```diff
@@ -489,7 +489,9 @@ private void checkClassAccess(Class<?> targetClass) throws IllegalAccessExceptio
 					return;
 				}
 			} else {
-				if (((PACKAGE == (accessMode & PACKAGE)) || Modifier.isPrivate(accessMode)) && isSamePackage(accessClass, targetClass)) {
+				if (((PACKAGE == (accessMode & PACKAGE))
```

Comment:
```
if (((PACKAGE == (accessMode & PACKAGE)) || Modifier.isPrivate(accessMode)) 
&& (isSamePackage(accessClass, targetClass) || (accessClass == targetClass))
) {
```
Splitting the conditions this way - were the bracketing and the line breaks agree, is clearer to read.

Can you do the `accessClass == targetClass` check before the `isSamePackage` check?  The first is a very cheap pointer comparison while the second is a more expensive check.

Reply:
Agreed and fixed.